core/types/header.go

```diff
@@ -138,6 +150,16 @@ func (bd *Header) deserializationUnsigned(source *common.ZeroCopySource) error {
 	var irregular, eof bool
 
 	bd.Version, eof = source.NextUint32()
+	if eof {
+		return io.ErrUnexpectedEOF
+	}
+	if bd.Version > CURR_HEADER_VERSION {
+		return common.ErrIrregularData
+	}
+	if bd.Version == VERSION_SUPPORT_SHARD {
```

Comment:
it's better: 
bd.Version >= VERSION_SUPPORT_SHARD

similar with transaction version check.